File: kerbianbridge/android/BridgeEventRouter.java
package kerbianbridge.android;

import org.json.JSONObject;

public class BridgeEventRouter {
    public static NativeEventSink eventSink;

    public static void emit(String event, int viewId, JSONObject data) {
        try {
            if (eventSink != null) {
                JSONObject msg = new JSONObject();
                msg.put("type", "event");
                msg.put("event", event);
                msg.put("view_id", viewId);
                msg.put("data", data != null ? data : new JSONObject());
                eventSink.send(msg.toString());
            }
        } catch (Exception e) {
            ErrorReporter.report("BridgeError", "Failed to emit event: " + event, new JSONObject().put("exception", e.toString()));
        }
    }
}

File: kerbianbridge/ios/BridgeEventRouter.m
#import "BridgeEventRouter.h"
#import "ErrorReporter.h"

static id _eventSink = nil;

@implementation BridgeEventRouter
+ (void)setEventSink:(id)sink { _eventSink = sink; }
+ (void)emit:(NSString *)event viewId:(NSInteger)viewId data:(NSDictionary *)data {
    @try {
        if (_eventSink && [_eventSink respondsToSelector:@selector(send:)]) {
            NSDictionary *msg = @{
                @"type": @"event",
                @"event": event,
                @"view_id": @(viewId),
                @"data": data ?: @{}
            };
            NSData *jsonData = [NSJSONSerialization dataWithJSONObject:msg options:0 error:nil];
            NSString *jsonLine = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
            [_eventSink send:jsonLine];
        }
    }
    @catch (NSException *e) {
        [ErrorReporter report:@"BridgeError" message:[NSString stringWithFormat:@"Failed to emit event: %@", event] details:@{@"exception": e.reason ?: @""}];
    }
}
@end